### ISSUE TITLE
Default to standard mode when compatible theme/plugins active

### ISSUE THREAD

**Issue by westonruter** [ampproject/amp-wp#1384]
~For brand new plugin activations, we should consider to default to the new Transitional mode.~ (As of v2.0, Transitional mode is the default when an AMP-compatible theme is active.) This would be instead of defaulting to the Reader mode, which is what the default would continue to be for existing users of the plugin when they upgrade. For new users, the Transitional mode can be much superior to the Reader mode because not only are same theme templates being used, but also any validation errors are actually reported to the user. ~In the Reader mode, validation errors are silently sanitized which is dangerous since it can mean ads and other critical site functionality is removed.~ (Not as of #4480.)

We should also consider defaulting to the new Transitional mode if the user has an active theme which is known to be compatible with AMP (e.g. the core themes). We can also consider the same for compatible. plugins. We could also consider doing this by default if no `excessive_css` validation errors are raised, as this should mean that the paired mode templates will at least be stylistically the same as the canonical templates.

This would hopefully increase the adoption of the new AMP theme support.

See also #1288.

**Comment by westonruter** [comment]
We could also show a deprecated notice to the users of the Classic mode to get them to switch to the new Paired mode. However, we should consider only doing this if the active theme does _not_ contain an `amp` directory, since this directory entails that they have a customized theme for the classic mode and thus probably they don't have a blue bar. 

Then in the future we can even remove Classic mode from being presented to users entirely. Similar to Links/Blogroll in WordPress, which are only available for grandfathered sites that had it enabled in the past, we can similarly hide the Classic option for new plugin activations in the future.

**Comment by postphotos** [comment]
As we've talked about a bunch, the "blue bar" classic experience isn't that great, especially with the leaps and bounds of functionality we've added in v0.7 and v1.0. 

Further, there's [this description](https://searchengineland.com/google-will-require-amp-canonical-pages-match-february-1-2018-286921) of AMP + Google Search from Search Engine Land:
> Though AMP isn't a ranking signal, non-compliant pages won't appear in Top Stories or other AMP-specific slots in search results.

Paired mode more closely meets this "canonical AMP" requirement, and Native AMP does it to 100%.

**Comment by alcurrie** [comment]
Based on the discussion in #1288 re.   I've added a new issue What to do with sites who run Classic Mode who can run Paired or Native modes in v1.1. #1538 - there may be some overlap with this discussion though.

**Comment by westonruter** [comment]
When compatible themes and plugins are active, we should eventually default to starting with Native mode. See also #2313 for guiding users to AMP-compatible themes/plugins.

**Comment by westonruter** [comment]
Note: As of 2.0, Transitional mode is now the default for new activations that have an AMP-compatible theme.

**Comment by jwold** [comment]
* WR > Blocked by site scan, and needs compatibility database, to default to standard mode. It will start in reader, do scan, then recommend a mode. Do scan and onboarding says results of site scan to say "site is compatible with Standard mode," try switching. 
* WR > Have sample set of URLs being scanned by WP Chron. If those results return with no errors we can be reasonable confident that site is AMP compatible. Site scan is triggered when you activate AMP plugin. Then shortly thereafter you should have validation results in system, wouldn't have to wait for scan to complete, it's possible they could be available and disabled in settings and in onboarding. 
* WR > If results aren't ready (when in onboarding or settings) we'll say "please wait, getting results," then we'd show them after. 
* AM > Could compatibly database inform scanning? 
* WR > Yes, two pieces. Site scanning is one half of the coin and then compatibility database is the other half. 
* AS > When you install and activate plugin it scans. When you go into wizard, or at the latest at the screen where you need results, it could do accelerated chron, or switch into synchronous mode to give results of 10 scans and show progress bar. Best case is it's already there when user comes to screen. Worst case is there's half work left, and it switches to doing as fast as possible when you get to that screen. 
* **Async scanning in general but sync only on onboarding.** 
* WR > If scanning hasn't finished then we'd show spinner saying it's scanning, or progress bar, and then initiate remaining scans in synchronous mode. 
* WR > Only scanning sample set of say 10 urls for this

* AS > What happens after when validation errors found after user was suggested to use standard mode? Should store that standard mode was suggested base on site scan and give extra hints if it was wrong choice after. 
* WR > Similar as well to if you install an AMP incompatible plugin later, we need to let them know that they were compatible, but not now and here's why. Need site health or proactive notice to user to say issue with AMP health. Site Chron is why. 
* AS > Yes, thinking of scenario where you're picking default and recommendation is standard, not initiated by you changing from something to standard. Should handle that separately with additional info. 
* AM > Plugin should still suggest, but not choose. You pick. 
* JW> No matter what scan says, let user pick and explain why. 
* AM > Given current state of site, from this result standard mode applies. But if you use incompatible. 
* AM > Hard to capture the state in a sentence. We just need to have right documentation specifying what this means. 
* JW > We need to have great UX writing to explain the state of what's going on. 
* WR > Notification for action items: https://github.com/ampproject/amp-wp/issues/5671. 

Dashboard

**Comment by westonruter** [comment]
So we shouldn't default to Standard mode when the plugin is activated in the same way we are defaulting to Reader mode, since we need the results from Site Scanning to make that decision. We also need the user input, for example if the user is not technically-savvy it may not be recommended for them to go down the Standard mode path.
